mypy/nodes.py

```diff
@@ -778,6 +780,9 @@ def __init__(
         self.is_mypy_only = False
         self.dataclass_transform_spec: DataclassTransformSpec | None = None
         self.docstring: str | None = None
+        # track the end of the function definition itself
+        self.def_end_line: int | None = None
+        self.def_end_column: int | None = None
```

Comment:
I think the new attributes will also need to be serialized, or they may get lost in incremental mode.

Reply:
I checked in `nodes.py` and no node serializes `end_line` or `end_column`. The same for `line` and `column` with the exception of the `TypeAlias` node (it's unclear to me why).

I did a quick test invoking mypy with `--incremental --disallow-untyped-defs --show-error-end` on the same file twice which resulted in consistent results.